code/controllers/subsystem/starmap.dm

```diff
@@ -149,6 +179,22 @@ SUBSYSTEM_DEF(starmap)
 
 		from_time = 0
 		to_time = 0
+
+		in_transit_planet = 0
+
+		sleep(1)
+
+		for(var/area/shuttle/ftl/F in world)
+			F << 'sound/effects/hyperspace_end.ogg'
+		toggle_ambience(0)
+
+		spawn(50)
+			ftl_sound('sound/ai/ftl_success.ogg')
+
+		if(current_planet.visited != TRUE)
+			current_planet.event.activate_event()
+			current_planet.visited = TRUE
```

Comment:
Put visited under the conditional block of line 161. You don't need to check if it's not true, just set it to TRUE and be done with it.